tests/std/tests/P0896R4_stream_iterators/test.cpp

```diff
@@ -101,6 +107,112 @@ void test_ostreambuf_iterator(basic_ostream<CharT, Traits>& os) {
 #endif // __cpp_lib_concepts
 }
 
+template <class C, enable_if_t<is_same_v<C, char>, int> = 0>
+constexpr auto statically_widen_impl(const char* narrow_str, const wchar_t*) noexcept {
+    return narrow_str;
+}
+
+template <class C, enable_if_t<!is_same_v<C, char>, int> = 0>
+constexpr auto statically_widen_impl(const char*, const wchar_t* wide_str) noexcept {
+    return wide_str;
+}
+
+#define STATICALLY_WIDEN(C, S) statically_widen_impl<C>(S, L##S)
+
+template <class Sequence>
+void test_regex_iterator(const Sequence& seq) {
+    (void) seq;
+#ifdef __cpp_lib_concepts
+    using SequenceIter = decltype(cbegin(seq));
+    using CharT        = remove_cv_t<iter_value_t<SequenceIter>>;
+    using I            = regex_iterator<SequenceIter>;
+
+    const I end_it{};
+    assert(end_it == default_sentinel);
+    assert(default_sentinel == end_it);
+    assert(!(end_it != default_sentinel));
+    assert(!(default_sentinel != end_it));
+
+    basic_regex<CharT> re(STATICALLY_WIDEN(CharT, "[a-z]"));
+    const I begin_it{cbegin(seq), cend(seq), re};
+    assert(begin_it != default_sentinel);
+    assert(default_sentinel != begin_it);
+    assert(!(begin_it == default_sentinel));
+    assert(!(default_sentinel == begin_it));

```

Comment:
It'd be nice to add a test for "iterator which is not default-initialized, but which is equal to `end()`":

```suggestion
    assert(!(default_sentinel == begin_it));

    advance(begin_it, end_it);
    assert(begin_it == default_sentinel);
    assert(default_sentinel == begin_it);
    assert(!(begin_it != default_sentinel));
    assert(!(default_sentinel != begin_it));
```

(same below)

Reply:
Done with `std::ranges::advance`. (`std::advance` is unlikely useful for this.)